src/uiSelectDirective.js

```diff
@@ -68,6 +68,18 @@ uis.directive('uiSelect',
             $select.sortable = sortable !== undefined ? sortable : uiSelectConfig.sortable;
         });
 
+        if(angular.isDefined(attrs.inputSearchModel)){
+          $select.search = scope.$parent.$eval(attrs.inputSearchModel);
+
+          scope.$parent.$watch(attrs.inputSearchModel, function(){
+            $select.search = scope.$parent.$eval(attrs.inputSearchModel);
+          });
+
+          scope.$watch('$select.search', function(){
+             scope.$parent.$eval(attrs.inputSearchModel + '= "' + $select.search + '"');
```

Comment:
A better approach is above before the $watch, cache via `var foo = $parse(attrs.inputSearchModel + '= "' + $select.search + '"')` and inside the watch listener callback apply scope.$parent, i.e. `foo(scope.$parent)`, with obviously better variable names.